covid19_sfbayarea/data/marin_scraper.py

```diff
@@ -0,0 +1,303 @@
+	"""This method gets the breakdown of cases and deaths by gender."""
+	csv_ = extract_csvs(chart_id, url)
+
+	csv_strs = csv_.splitlines()
+	keys = csv_strs[0].split(',') 
+	if keys != ['Gender', 'POPULATION', 'Cases', 'Hospitalizations', 'Deaths']:
+		raise ValueError('The headers have changed.')
+
+	genders = ['male', 'female']
+	c_gender = {}
+	d_gender = {}
+	
+	for row in csv_strs[1:]:
+		# Extracting the gender and the raw count (the 3rd and 5th columns, respectively) for both cases and deaths.
+		# Each new row has data for a different gender.
+		split = row.split(',')
+		gender = split[0].lower()
+		if gender not in genders:
+			return ValueError('The genders have changed.')
+		c_gender[gender] = int(split[2])
+		d_gender[gender] = int(split[4])			
+
+	return c_gender, d_gender
+
+def get_breakdown_race_eth(chart_id: str, url: str) -> Tuple:
+	"""This method gets the breakdown of cases and deaths by race/ethnicity."""
+
+	csv_ = extract_csvs(chart_id, url)
+
+	csv_strs = csv_.splitlines()
+	keys = csv_strs[0].split(',') 
+	
+	if keys != ['Race/Ethnicity', 'COUNTY POPULATION', 'Case Count', 'Percent of Cases', 'Hospitalization Count', 'Percent of Hospitalizations', 'Death Count', 'Percent of Deaths']:
+		raise ValueError("The headers have changed.")
+
+	key_mapping = {"black/african american":"African_Amer", "hispanic/latino": "Latinx_or_Hispanic",
+            "american indian/alaska native": "Native_Amer", "native hawaiian/pacific islander": "Pacific_Islander", "white": "White", "asian": "Asian", "multi or other race": "Multi or Other Race"}
+            # "Multiple_Race", "Other" are not separate in this data set - they are one value under "Multi or Other Race"
+
+	c_race_eth = {}
+	d_race_eth = {}
+	
+	for row in csv_strs[1:]:
+		split = row.split(',')
+		race_eth = split[0].lower()
+		if race_eth not in key_mapping:
+			raise ValueError("The race_eth groups have changed.")
+		else:
+			c_race_eth[key_mapping[race_eth]] = int(split[2])
+			d_race_eth[key_mapping[race_eth]] = int(split[6])
+
+	return c_race_eth, d_race_eth
+
+def get_test_series(chart_id: str, url: str) -> Tuple:
+	"""This method gets the date, the number of positive and negative tests on that date, and the number of cumulative positive and negative tests."""
+
+	csv_ = extract_csvs(chart_id, url)
+	series = []
+
+	csv_strs = csv_.splitlines()
+	keys = csv_strs[0].split(',')
+	
+	test_history = []
+	
+	# Grab the dates, which are in the header
+	for entry in csv_strs[:1][0].split(',')[1:]: 
+		# need to exclude very first item in the csv_strs[:1][0].split(',') list (which is the value 'Date')
+		daily = {}
+		date_time_obj = datetime.strptime(entry, '%m/%d/%Y')
+		daily["date"] = date_time_obj.isoformat()
+		series.append(daily)
+
+	# The slicing makes this if statement hard to look at... there must be a better way?
+	if csv_strs[1:2][0].split(',')[:1][0] != 'Positive Tests' and csv_strs[2:][0].split(',')[:1][0] != 'Negative Tests':
```

Comment:
If you’re just trying to get a single entry from a list, you can just do `list[index]`. e.g:

```py
csv_strs[1:2][0]
# Is the same as:
csv_strs[1]

# And:
csv_strs[1].split(',')[:1][0]
# Is the same as:
csv_strs[1].split(',')[0]
```

Reply:
that makes sense LOL. i don't know why i was writing such strange code... thank you!